fix: _VER variables break yarn

blimmer opened the pull request:
The new `cimg/node:lts`, which points at 16.15, breaks `yarn` berry:

```
 docker run -it --rm --entrypoint bash cimg/node:lts
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
circleci@9fdbb66acc23:~/project$ yarn set version berry
Usage Error: Unrecognized or legacy configuration settings found: ver - run "yarn config -v" to see the list of settings supported in Yarn (in <environment>)

$ yarn set version [--only-if-needed] <version>
circleci@9fdbb66acc23:~/project$ unset YARN_VER
circleci@9fdbb66acc23:~/project$ yarn set version berry
➤ YN0000: Retrieving https://repo.yarnpkg.com/3.2.0/packages/yarnpkg-cli/bin/yarn.js
➤ YN0000: Saving the new release in .yarn/releases/yarn-3.2.0.cjs
➤ YN0000: Done in 2s 453ms
```

In all other Dockerfiles, you're using `YARN_VERSION`, not `YARN_VER`. Since this breaks all yarn v2+ (berry) users, I think it's worth publishing over the existing image.

hasghari commented:
Just ran into this as well and we're blocked as a result.

blimmer commented:
@hasghari - if you're able, you can change `cimg/node:lts` to `cimg/node:16.14.2` to temporarily work around this. Also, if you have enterprise support, you might reach out to CircleCI support to see if they can get eyes on this internally.

johnellmore commented:
Also experiencing this issue. My team is blocked as well.

hasghari commented:
> @hasghari - if you're able, you can change `cimg/node:lts` to `cimg/node:16.14.2` to temporarily work around this. Also, if you have enterprise support, you might reach out to CircleCI support to see if they can get eyes on this internally.

Thanks but we've locked the version to `16.15.0` to match production. We will have to downgrade to `16.14.2` if there's no speedy resolution here.

JalexChen commented:
Hi @blimmer,

Thanks for bringing this to our attention. in the essence of time, given so many people seem to be affected and blocked, I went into your PR and made the necessary changes so you are still getting credit for bringing this up.

1) First thing is first - please make the change in the Dockerfile.template, as this generates all the other dockerfiles through ./shared/gen-dockerfiles <insert node versions here separated by space>

2) NODE_VERSION is a variable that the install.sh script uses, so there's some significance there, it was the intention of shortening it as part of a new pattern we're implementing, and the explicit nvm install NODE_VER was to make it crystal clear, but we can change this back at a later date

3) YARN_VERSION was reverted back from it's short form

4) there were a couple of variables that did not get changed to match so I changed those for you as well

JalexChen reviewed (approved):
Hi @blimmer - Again, thanks for bringing this up. Would love for you to take a look at the comment I made in the issue, but I think we should be good to go here. Appreciate you taking the time to put in the PR!

hasghari commented:
@JalexChen Thanks for the fix. When can we expect the updated images to be available?

JalexChen commented:
@hasghari could you check right now to see? the new images should be up right now

hasghari commented:
> @hasghari could you check right now to see? the new images should be up right now

Yes, I can confirm the image is fixed. Thanks!

blimmer commented:
@JalexChen - can you please republish the `lts` tags?

![Screen Shot 2022-05-12 at 07 52 12](https://user-images.githubusercontent.com/630449/168091263-fbefb55f-0dae-43c4-9ce5-94f3a628bca2.png)

JalexChen commented:
hi @blimmer i believe these lts and current were built and pushed. Could you please check to confirm?

blimmer commented:
@JalexChen - yep, it looks like they were published recently to DockerHub. I confirmed that `lts` is working properly on my machine:

```
 ~  docker image rm cimg/node:lts
Untagged: cimg/node:lts
Untagged: cimg/node@sha256:49e010b94427e077b111f52b7777bdcae56a0e98c778a0fd0a5bf146a957fc36
Deleted: sha256:5f6ca43ffdae61e58025aaa1e7a0ddd1728a7d82b5542cb722d38c8225abfe85
Deleted: sha256:d0a6f4d9f4a4116e289cd9a1a26f9ded1d295a2d795974080c9307347e219149
Deleted: sha256:e124b28f75bdd838b23bc8f362de31cb3b3a53ec2a5beb9430750f2e2e31b17a
 ~  docker pull cimg/node:lts
lts: Pulling from cimg/node
e0b25ef51634: Already exists
412d105bb1ee: Already exists
45a96c2a2995: Already exists
284809b37068: Already exists
38c14eae51d3: Already exists
5606f68b4252: Already exists
85a817ede446: Pull complete
72cf2094c33b: Pull complete
Digest: sha256:09659be76126b5b6a0cd332485d61d431561eba4f9344a3385fbc96255d0121b
Status: Downloaded newer image for cimg/node:lts
docker.io/cimg/node:lts
 ~  docker run -it --rm --entrypoint bash cimg/node:lts
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
circleci@0596be1e6df2:~/project$ yarn set version berry
➤ YN0000: Retrieving https://repo.yarnpkg.com/3.2.0/packages/yarnpkg-cli/bin/yarn.js
➤ YN0000: Saving the new release in .yarn/releases/yarn-3.2.0.cjs
➤ YN0000: Done in 2s 487ms
```

Thanks!

blimmer commented:
Also, I wonder, is it possible to add an integration test step to this project that does things like:

- `npm ci`
- `npm install`
- `yarn install` (v1)
- `yarn install` (berry)

to try to prevent this kind of thing in the future?

tjmcs commented:
Sounds like a great idea, @blimmer; @felicianotech and @JalexChen let's make sure we add that to our discussion next week :)
